kairos-server/src/config.rs

```diff
@@ -1,26 +1,119 @@
-use std::{fmt, str::FromStr};
+use config::{Config as Configuration, Environment, File};
+use dotenvy::dotenv;
+use serde::{Deserialize, Serialize};
+use std::{fs::File as FsFile, net::SocketAddr, path::Path};
+use tracing::{subscriber::set_global_default, Level};
+use tracing_subscriber::{fmt, prelude::*, EnvFilter};
 
-#[derive(Debug)]
-pub struct ServerConfig {
-    pub port: u16,
-}
+fn serialize_level_filter<S>(level: &Level, serializer: S) -> Result<S::Ok, S::Error>
```

Comment:
there is a crate called tracing-serde https://github.com/tokio-rs/tracing/tree/master/tracing-serde

Reply:
Thank you! Can't believe I missed this.

Reply:
Turns out this is for serializing logs to a machine readable format. Not useful for serializing/deserialzing the log level.